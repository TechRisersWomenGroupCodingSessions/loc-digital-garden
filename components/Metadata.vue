<script>
export default {
  props: {
    planted: {
      type: Number,
      required: true,
    },
    tended: {
      type: Number,
      required: true,
    },
  },
  data() {
      return {
          plantedAsDays: this.planted,
          tendedAsDays: this.tended,
      }
  },
};
</script>

<template>
  <div>
    <p>Planted: {{ plantedAsDays }} days ago</p>
    <p>Last tended: {{ tendedAsDays }} days ago</p>
  </div>
</template>
