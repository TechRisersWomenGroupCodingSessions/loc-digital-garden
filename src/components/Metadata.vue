<script>
export default {
  name: "EntryMetadata",
  props: {
    planted: {
      type: Number,
      required: true,
    },
    tended: {
      type: Number,
      required: true,
    },
  },
  data() {
    return {
      plantedAsDays: this.planted,
      tendedAsDays: this.tended,
    };
  },
};
</script>

<template>
  <div>
    <p>Planted: {{ plantedAsDays }} days ago</p>
    <p>Last tended: {{ tendedAsDays }} days ago</p>
  </div>
</template>
