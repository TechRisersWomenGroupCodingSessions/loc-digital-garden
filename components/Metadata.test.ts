import { describe, it, expect, beforeEach } from 'vitest'
import { shallowMount } from '@vue/test-utils' 
import Metadata from "./Metadata.vue"

describe('Metadata', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(Metadata, {
            props: {
                planted: 2,
                tended: 14,
            }
        })
    });
    it('renders as expected', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it.skip('contains expected content', () => {
        expect(wrapper.text()).toContain("Planted 2 days ago");
        expect(wrapper.text()).toContain("Tended 2 days ago");
    });
});