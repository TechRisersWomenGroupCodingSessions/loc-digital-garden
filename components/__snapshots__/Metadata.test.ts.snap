// Vitest Snapshot v1

exports[`Metadata > renders as expected 1`] = `
"<div>
  <p>Planted: 2 days ago</p>
  <p>Last tended: 14 days ago</p>
</div>"
`;
